tools/check_contrast.py

```
@@ -0,0 +1,160 @@
+
+def ratio(c1, c2):
+    l1, l2 = lum(hex_to_rgb(c1)), lum(hex_to_rgb(c2))
+    if l1 < l2:
+        l1, l2 = l2, l1
+    return (l1 + 0.05) / (l2 + 0.05)
+
+
+def mix(c1, c2, w1):
+    """color-mix in srgb: gamma-space channel interpolation."""
+    a, b = hex_to_rgb(c1), hex_to_rgb(c2)
+    return "#%02x%02x%02x" % tuple(round(w1 * x + (1 - w1) * y) for x, y in zip(a, b))
```

Ruff (0.16.1)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)